### PR TITLE
if user type a wrong technology name they will be guild by help of bot

### PR DESCRIPTION
**Pull request by JisanAR03**
if your type wrong tech name bot will provide a message preview

image

@arkid15r sir, please let me know , if there is any suggestion for me

### CODE REVIEW

**Review by arkid15r — CHANGES_REQUESTED** [review]
@JisanAR03 could you resolve the conflicts before I review it?

Thank you

**Comment by JisanAR03** [comment]
> @JisanAR03 could you resolve the conflicts before I review it?
> 
> Thank you

@arkid15r , just make another commit , can you please review ?

File: src/lettuce/plugins/repo.py [hunk 1]
```diff
@@ -21,11 +18,50 @@ def __init__(self, client: SlackClient, settings: CaseInsensitiveDict, storage:
...
+                    },
+                    {
+                        "type": "actions",
+                        "block_id": "tech_select_block",
+                        "elements": [
+                            {
+                                "type": "button",
+                                "text": {"type": "plain_text", "text": tech},
+                                "value": tech,
+                                "action_id": f"button_{tech}",
```
Comment (arkid15r) [inline src/lettuce/plugins/repo.py:47]: It seems this has a quite broad scope. Let's use more specific naming here:
```suggestion
                                "action_id": f"repo_plugin_button_{tech}",
```

File: tests/plugins/test_repo.py [hunk 2]
```diff
@@ -1,63 +1,86 @@
...
+        "Hello, you can implement your 'python' knowledge here:\nhttps://github.com/example/python-repo"
+    )
+
+
+@pytest.mark.asyncio
+async def test_repo_command_no_repo(repo_plugin, mock_command):
+    """Test the repo command with a nonexistent repository."""
+    mock_command.text.strip.return_value.lower.return_value = "nonexistent"
+    await repo_plugin.repo(mock_command)
+    assert mock_command.say.call_count == 0
```
Comment (arkid15r) [inline tests/plugins/test_repo.py:76]:
```suggestion
    mock_command.say.assert_not_called()
```

Comment (JisanAR03) [reply tests/plugins/test_repo.py]: @arkid15r sir  , can you check the updated code ?

File: src/lettuce/plugins/repo.py [hunk 3]
```diff
@@ -21,11 +18,50 @@ def __init__(self, client: SlackClient, settings: CaseInsensitiveDict, storage:
...
 
         repos = self.repo_data.get(tech_name)
         if repos:
             repos_list = "\n".join(repos)
             message = f"Hello, you can implement your '{tech_name}' knowledge here:\n{repos_list}"
+            await command.say(message)
         else:
-            message = f"Hello, the technology '{tech_name}' is not recognized. Please try again."
-        await command.say(message)
+            fallback_message = "Available technologies :"
```
Comment (arkid15r) [inline src/lettuce/plugins/repo.py:29]:
```suggestion
            fallback_message = "Available technologies:"
```

**Review by arkid15r — APPROVED** [review]

**Pull request merged**